river/ensemble/streaming_random_patches.py

```diff
@@ -399,8 +413,9 @@ def learn_one(
         *,
         sample_weight: int,
         n_samples_seen: int,
-        rng: np.random.RandomState,
+        rng: np.random.Generator,
     ):
+        # print(sample_weight)
```

Comment:
Thanks for the review @smastelini 

> Could you please add the changes to the Naive Bayes calculation within the trees' leaves?

I skipped this on purpose as the root cause is unrelated to the Hoeffding Trees. I will create a different PR for that change in order to isolate and analyze its impact across the project.

> What about making SRP cope with #420? I believe we are currently skipping those tests for SRP.

This PR is already closed. Moreover, supporting emerging features in `SRP` is non-trivial and the impact on performance is unknown, especially since feature selection is a key component of the `SRP` algorithm.

Reply:
Thanks for the review @smastelini 

> Could you please add the changes to the Naive Bayes calculation within the trees' leaves?

I skipped this on purpose as the root cause is unrelated to the Hoeffding Trees. I will create a different PR for that change in order to isolate and analyze its impact across the project.

> What about making SRP cope with #420? I believe we are currently skipping those tests for SRP.

This PR is already closed. Moreover, supporting emerging features in `SRP` is non-trivial and the impact on performance is unknown, especially since feature selection is a key component of the `SRP` algorithm.

Reply:
Maybe I forgot some important corner case, but as I see it, two strategies should be used to ensure robustness in case of missing/emerging features:

1. Keep a set of all observed features to sample from when creating the random subspaces
2. Use try/except statements to ensure errors do not stop the execution when accessing features from the subspaces in an "incomplete" instance.

Am I missing the point?

Reply:
> I skipped this on purpose as the root cause is unrelated to the Hoeffding Trees. I will create a different PR for that change in order to isolate and analyze its impact across the project.

Totally makes sense.